src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.cs

```diff
@@ -48,6 +48,79 @@ public JsonArray(params JsonNode?[] items) : base()
             InitializeFromArray(items);
         }
 
+        internal JsonNode DeepCloneArray()
+        {
+            if (_jsonElement.HasValue)
+            {
+                return new JsonArray(_jsonElement.Value.Clone(), Options);
+            }
+
+            var jsonArray = new JsonArray(Options);
+
+            for (int i = 0; i < List.Count; i++)
+            {
+                JsonNode? item = List[i];
+                if (item is null)
+                {
+                    jsonArray.Add(null);
+                }
+                else
+                {
+                    jsonArray.Add(item.DeepClone());
+                }
+            }
+
+            return jsonArray;
+        }
+
+        internal override bool DeepEquals(JsonNode? node)
+        {
+            if (node is null || node is JsonObject)
+            {
+                return false;
+            }
+
+            if (node is not JsonArray array)
+            {
+                return node.DeepEquals(this);
+            }
+
+            if (List.Count != array.List.Count)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < List.Count; i++)
+            {
+                JsonNode? currentItem = List[i];
+                JsonNode? otherItem = array.List[i];
+
+                if (!DeepEquals(currentItem, otherItem))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        internal int GetElementIndex(JsonNode? node)
+        {
+            return List.IndexOf(node);
+        }
+
+        /// <summary>
+        /// Returns enumerator that wraps calls to <see cref="JsonNode.GetValue{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of the value to obtain from the <see cref="JsonValue"/>.</typeparam>
+        public IEnumerable<T> GetValues<T>()
+        {
+            foreach (JsonNode? item in List)
+            {
+                yield return item!.GetValue<T>();
```

Comment:
This will throw NRE whenever a `null` element is encountered:
```suggestion
                yield return item is null ? (T)(object?)null! : item.GetValue<T>();
```

Reply:
Thanks I have added test cases for it https://github.com/dotnet/runtime/pull/87381/files#diff-df220dc578eb5731310c69519699ca63947384f836755cbea556748bc60aa3d4R648